src/pages/TestingPage/TestControls/controlReducer.ts

```diff
@@ -0,0 +1,102 @@
+import { Action, Form, FormData } from "./TestAttributes";
+
+const searchId = (form: Form, id: string): number => {
+    return form.formData.findIndex((inputData) => inputData.id == id);
+};
+
+const checkType = (type: string, value: number): boolean => {
+    //TODO: which types?
+    return true;
+};
+
+const checkIsValid = (formData: FormData): boolean => {
+    const result =
+        formData.reduce(
+            (prev, currentInput) =>
+                prev &&
+                (currentInput.validity.isValid || !currentInput.enabled),
+            true
+        ) &&
+        formData.some((input) => input.enabled) &&
+        !formData.some((input) => input.enabled && !input.value);
+    return result;
+};
+
+export const checkValidityInputs = (initialState: Form): Form => {
+    const newState = [...initialState.formData];
+    newState.forEach((input) => {
+        if (input.value && checkType(input.type, input.value)) {
+            input = {
+                ...input,
+                validity: { isValid: true, msg: input.validity.msg },
+            };
+        } else {
+            input = {
+                ...input,
+                validity: { isValid: false, msg: input.validity.msg },
+            };
+        }
+    });
+
+    return { formData: newState, isValid: checkIsValid(newState) };
+};
+
+export const taskReducer = (state: Form, action: Action): Form => {
+    switch (action.type) {
+        case "CHANGE_ENABLE": {
+            const dataIndex = searchId(state, action.payload.id);
+            const newFormData = [...state.formData];
+            let validity = true;
+            if (!action.payload.enabled) {
+                validity = false;
+            }
+
+            newFormData[dataIndex] = {
+                ...newFormData[dataIndex],
+                enabled: action.payload.enabled,
+                validity: {
+                    isValid: validity,
```

Comment:
Dont modify the validity of the field in this action, only the validity of the form